### ISSUE TITLE
A new "Show-more" pattern

### ISSUE THREAD

**Issue by solazio** [canonical-web-and-design/vanilla-framework#2815]
This would be used to wrap any tall `<tag>` and "truncate" it to a specific height. Once clicking the "Show more" button the full size of the `<tag>`.

Currently this pattern is used on snapcraft in 3 locations:
- https://snapcraft.io/#c
- https://snapcraft.io/install/riot-web/centos
- https://snapcraft.io/riot-web - distro list on mobile only

Link to codepen: https://codepen.io/solazio/pen/gOpYVqQ

**Comment by bartaz** [comment]
Extends on existing "back to top"

WG:
- use/modify existing back to top (change the name)
- have separate 'fadeout' pattern
- explore design

@danielmutis Provide examples where it works where it doesn't

**Comment by danielmutis** [comment]
### Examples of places where it works:

[Snapcraft first snap flow](https://snapcraft.io/first-snap#python)
![Screenshot 2020-02-26 at 14.01.28.png](https://images.zenhubusercontent.com/5e1c5367f4dbe2bfe4c77ad8/f4cfdcc0-c8c9-4d2d-9315-07061612fbc2)

[Snapcraft list of users by distro for a single snap:
](https://snapcraft.io/riot-web)![Screenshot 2020-02-26 at 14.02.37.png](https://images.zenhubusercontent.com/5e1c5367f4dbe2bfe4c77ad8/eef738e8-3598-447e-9d7f-c0c3ecda98bf)

### Example where it looks odd:

Builds table:
![Screenshot 2020-02-26 at 14.03.50.png](https://images.zenhubusercontent.com/5e1c5367f4dbe2bfe4c77ad8/838a4981-2a99-48be-936d-beacf3b180ea)

**Comment by matthewpaulthomas** [comment]
> Extends on existing "back to top"

I don’t know how much the design here intentionally follows “Back to top”. But things that act different should look different, and the behaviour of “Show more” is completely different from “Back to top”. So I think it’s misleading for them to look as similar as they do.

Interaction-wise, this is bog-standard [progressive disclosure](https://www.givegoodux.com/the-power-of-progressive-disclosure/). Vanilla has an existing element for progressive disclosure, though it’s not currently described: [`<details>`](https://vanillaframework.io/docs/examples/base/details).

![](https://user-images.githubusercontent.com/19801137/76326014-eeb09380-62df-11ea-865a-49ef1dc38052.png)

And Vanilla sites currently use [a variety of icons for progressive disclosure](https://github.com/canonical-web-and-design/web-squad/issues/2411), for example “+” and “–” in MAAS:

![](https://user-images.githubusercontent.com/19801137/75545972-862fff80-5a1f-11ea-9c55-dd887b716944.png)

Now, we don’t have to use the same icon or style for this everywhere. For example, GitHub uses [three different icons for progressive disclosure in different layout situations](https://primer.style/design/ui-patterns/progressive-disclosure). What I _do_ think we need, if we’re going to have more than one, is an explanation of when to use each one.

> [Snapcraft first snap flow:](https://snapcraft.io/first-snap#python)
![Screenshot 2020-02-26 at 14.01.28.png](https://images.zenhubusercontent.com/5e1c5367f4dbe2bfe4c77ad8/f4cfdcc0-c8c9-4d2d-9315-07061612fbc2)

That probably isn’t a good example to base a pattern on, for two reasons. First, it’s usually hiding ten lines or fewer, so it’s probably consuming more time in clicking than it saves in scrolling. And second, the purpose of that block is to show off how short and simple the Yaml is, so hiding part of it by default is defeating the point.

> ### Example where it looks odd
> Builds table:

I don’t understand the purpose of this one either, since there’s nothing below that table except the footer, so you won’t be trying to scroll past it. That aside, can you elaborate on why it looks odd? Some possibilities:
- There’s nothing visual to indicate it’s interactive! There’s the word “Show”, but no border, no colour, no icon, no underline. What’s the visually simplest way of making it look interactive? (Whatever it is, you might find that you don’t need the word “Show” any more.)
- Relatedly, it’s the same size, colour, and alignment as a column of text above it that isn’t interactive. Again, things that act different should look different.
- Either there’s no fadeout, or the fadeout is invisibly applying to whitespace below the table.

In short, I think this needs:
- [ ] A description of when you should use progressive disclosure at all.
- [ ] A description of when you should use this, vs. `<details>`, vs. something else.
- [ ] Either a design change to resolve the oddness with the builds table, or a guideline that it shouldn’t be used for tables.

**Comment by bartaz** [comment]
We could use the new 'Show more' styling used in modular table, with an addition of faded background.

We should propose a new container component that uses the ones mentioned above.

**Comment by bartaz** [comment]
This have been through WG before, but because it needs UX work we should pass it though our new process.

**Comment by clagom** [comment]
@elioqoshi to investigate this and look at the modular table spec for existing solutions.

**Comment by clagom** [comment]
@SirSamTumless ready for planning

**Comment by clagom** [comment]
WG: keeping it simple, this should just be a blue link (related to the command link discussion) without lines, icons or any other elements.

**Comment by clagom** [comment]
WG: on the link area in Vanilla include this "show-more" option recommending using the command-link: https://vanillaframework.io/docs/patterns/links

**Comment by clagom** [comment]
Work in progress here: https://app.zenhub.com/workspaces/-design-system-tribe-59438c953747487777fbe484/issues/canonical-web-and-design/vanilla-framework/3923
